---
title: "RNA-Seq Analysis 5.24.23"
author: "Brandon Roy"
date: "2023-05-25"
output: html_document
---

```{r}
#import and load relevant packages

library(DESeq2) # main RNA-Seq analysis pipeline
library(pheatmap)
library(PCAtools)
library(tidyverse)
library(ggplot2)
library(ggfortify)
library(goseq)
library(ggpubr)
library(tidyr)
library(dplyr)
library(WGCNA) # network analysis
library(wesanderson)
library(Rmisc)
library(grid)
library(gridExtra)
library(lemon)
library(purrr)
library(rnaseqGene)
library(IHW)
library(ggrepel)
library(multiClust) # generating hierarchal clustering
#library(glmGamPoi)
library(pcaExplorer)
library(iSEE)
library(factoextra)

```

### Perform this code block once to gather gene counts for all samples
``{r setup, include=FALSE}

# assemble gene counts from featureCounts into single matrix
f_files<- list.files("./featureCounts output", full.names = T)
print(f_files)
read_in_feature_counts<- function(file){
  cnt<- read_tsv(file, col_names=T, comment='#')
  cnt<- cnt %>% dplyr::select(-Chr, -Start, -End, -Strand, -Length)
  return(cnt)
}
raw_counts<- map(f_files, read_in_feature_counts)
raw_counts_df<- purrr::reduce(raw_counts,inner_join)

# all counts are in one matrix and can be manipulated # now write to file
write.table(raw_counts_df, file = "geneCount.txt", row.names = F, col.names = F)
raw_counts_df <- read.table(file= "geneCount.txt")
``


```{r}

#import necessary files
metadata <- read.table("Sample_metadata.txt", header = TRUE)

matrixFile <- "geneCount.txt"
sampleFile <- "Sample_metadata.txt"

#Input file manipulation
originalmatrixFile <- read.delim(matrixFile,header=FALSE,sep = " ")
cleanedmatrixFile <- originalmatrixFile[!duplicated(originalmatrixFile$V1), ]
cts1 <- data.frame(cleanedmatrixFile[,-1], row.names=cleanedmatrixFile[,1])

cts <- cts1[,-18]

coldata1 <- read.delim("Sample_metadata.txt", sep='\t',row.names=1,header=TRUE )
coldata <- coldata1[-18,]
colnames(cts) <- rownames(coldata)

#Test whether rownames of coldata match colnames of cts
all(rownames(coldata) == colnames(cts)) #should return true

# Adding a column 'group' to represent interaction of treatment*time:
coldata$group <- factor(paste0(coldata$Treatment, coldata$Time))
coldata$Time <- as.factor(coldata$Time)
coldata$Treatment <- as.factor(coldata$Treatment)
ggplot(data=cts, )
barplot(colSums(cts[,1:26]))
sum(colSums(cts[,1:26]))/26
```

```{r}
# Deseq2 model construction: (design = ~ Treatment + Time + Interaction)
dds <- DESeqDataSetFromMatrix(countData = cts, colData = coldata, design = ~1 + group)
keep <- rowSums(counts(dds)) >= 20
dds <- dds[keep,]

# Run DeSeq2
dds_out <- DESeq(dds)
print("Group comparison outputs of DESeq2 model:")
print(resultsNames(dds_out))
print("Dispersion Estimates of Counts")
plotDispEsts(dds_out)

# Transformation
vsd <- vst(dds_out, nsub=1000)
vsd
# plotDispEst(vsd)
cts_vst <- assay(vsd)
cts_vst <- as.data.frame(t(cts_vst))
datExpr0 <- as.matrix(cts_vst)

despca <- plotPCA(vsd, intgroup= c("Treatment","Time"), returnData=T)
ggplot(despca, aes(x=PC1, y=PC2, color = as.factor('Treatment'), shape=as.factor("Time"))) +
  geom_point(color = c('gray','gray','gray','orange','orange','orange','red','red','red','gray','gray','gray','orange','orange','orange','red','red','gray','gray','gray','orange','orange','orange','red','red','red'), pch = c(15,15,15,15,15,15,15,15,15,16,16,16,16,16,16,16,16,17,17,17,17,17,17,17,17,17), size = 5) + xlab("") + ylab("") + geom_point(shape = c(0,0,0,0,0,0,0,0,0,1,1,1,1,1,1,1,1,2,2,2,2,2,2,2,2,2),size = 5,colour = "black")

pca_res1 <- prcomp(datExpr0, scale. = F)
autoplot(pca_res1, label=T)


pcascree(p, type=c("pev","cev"))
rv <- rowVars(assay(vsd))
select <- order(rv, decreasing=TRUE)[seq_len(min(500,length(rv)))]
pca <- prcomp(t(assay(vsd)[select,]))
percentVar <- pca$sdev^2 / sum(pca$sdev^2)
scree_plot=data.frame(percentVar)
scree_plot[,2]<- c(1:26)
colnames(scree_plot)<-c("variance","component_number")
ggplot(scree_plot, mapping=aes(x=component_number, y=variance))+geom_bar(stat="identity")
  with(scree_plot, plot(component_number,variance), pch=20, xlim=c(0,10))+geom_bar(stat="identity")
```

```{r}
rownames(datExpr0) <- vsd$group

sampleTree <- hclust(dist(datExpr0), method = "mcquitty") # This method seems to group the data the best, weighted pair cluster analysis
plot(sampleTree)
d <- dist(datExpr0)
hc <- hclust(d)
plot(as.dendrogram(hc))
rect.hclust(hc, k=15, which = 15)
```

```{r}


ghuwtvsmut04 <- data.frame(results(dds_out, contrast=c("group", "wildtype_GHu4", "mutant_GHu4"),alpha=0.05), na.rm=T)
ghuwtvsmut04 <- na.omit(ghuwtvsmut04, cols=ghuwtvsmut26$padj)
sum(ghuwtvsmut04$padj < 0.05 & ghuwtvsmut04$log2FoldChange > 1)
sum(ghuwtvsmut04$padj < 0.05 & ghuwtvsmut04$log2FoldChange < -1)
ggplot(data=ghuwtvsmut04, aes(x=log2FoldChange, y=-log10(pvalue), col=ghuwtvsmut04$diffexpressed, label=ghuwtvsmut04$delabel)) + 
    geom_point() + 
    theme_minimal()
with(ghuwtvsmut04, plot(log2FoldChange, -log10(pvalue), pch=20, main="wildtype GHu vs mutant GHu 1EK802G (4dpi)", xlim=c(-7,9), ylim=c(0,10)))+
  with(subset(ghuwtvsmut04, pvalue<.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="light blue"))+
  with(subset(ghuwtvsmut04, abs(log2FoldChange)>1), points(log2FoldChange, -log10(pvalue), pch=20, col="gray"))+
  with(subset(ghuwtvsmut04, pvalue<.05 & abs(log2FoldChange)>1), points(log2FoldChange, -log10(pvalue), pch=20, col="light green"))


ghuwtvsmut17 <- data.frame(results(dds_out, contrast=c("group", "wildtype_GHu17", "mutant_GHu17"),alpha=0.05), na.rm=T)
ghuwtvsmut17 <- na.omit(ghuwtvsmut17, cols=ghuwtvsmut17$padj)
sum(ghuwtvsmut17$padj < 0.05 & ghuwtvsmut17$log2FoldChange > 1)
sum(ghuwtvsmut17$padj < 0.05 & ghuwtvsmut17$log2FoldChange < -1)
ggplot(data=ghuwtvsmut17, aes(x=log2FoldChange, y=-log10(pvalue), col=ghuwtvsmut17$diffexpressed, label=ghuwtvsmut17$delabel)) + 
    geom_point() + 
    theme_minimal()
with(ghuwtvsmut17, plot(log2FoldChange, -log10(pvalue), pch=20, main="wildtype GHu vs mutant GHu 1EK802G (17dpi)", xlim=c(-15,15), ylim=c(0,10)))+
  with(subset(ghuwtvsmut17, pvalue<.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="light blue"))+
  with(subset(ghuwtvsmut17, abs(log2FoldChange)>1), points(log2FoldChange, -log10(pvalue), pch=20, col="gray"))+
  with(subset(ghuwtvsmut17, pvalue<.05 & abs(log2FoldChange)>1), points(log2FoldChange, -log10(pvalue), pch=20, col="light green"))

ghuwtvsmut26 <- data.frame(results(dds_out, contrast=c("group", "wildtype_GHu26", "mutant_GHu26"),alpha=0.05), na.rm=T)
ghuwtvsmut26 <- na.omit(ghuwtvsmut26, cols=ghuwtvsmut26$padj)
sum(ghuwtvsmut26$padj < 0.05 & ghuwtvsmut26$log2FoldChange > 1)
sum(ghuwtvsmut26$padj < 0.05 & ghuwtvsmut26$log2FoldChange < -1)
ggplot(data=ghuwtvsmut26, aes(x=log2FoldChange, y=-log10(pvalue), col=ghuwtvsmut26$diffexpressed, label=ghuwtvsmut26$delabel)) + 
    geom_point() + 
    theme_minimal()
with(ghuwtvsmut26, plot(log2FoldChange, -log10(pvalue), pch=20, main="wildtype GHu vs mutant GHu 1EK802G (26dpi)", xlim=c(-7,9), ylim=c(0,10)))+
  with(subset(ghuwtvsmut26, pvalue<.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="light blue"))+
  with(subset(ghuwtvsmut26, abs(log2FoldChange)>1), points(log2FoldChange, -log10(pvalue), pch=20, col="gray"))+
  with(subset(ghuwtvsmut26, pvalue<.05 & abs(log2FoldChange)>1), points(log2FoldChange, -log10(pvalue), pch=20, col="light green"))


```

```{r}

#> print(resultsNames(dds_out))
#[1] "Intercept"                         "group_control26_vs_control17"      "group_control4_vs_control17"      
#[4] "group_mutant_GHu17_vs_control17"   "group_mutant_GHu26_vs_control17"   "group_mutant_GHu4_vs_control17"   
#[7] "group_wildtype_GHu17_vs_control17" "group_wildtype_GHu26_vs_control17" "group_wildtype_GHu4_vs_control17" 

ghuwtvsctrl04 <- data.frame(results(dds_out, contrast=c("group", "wildtype_GHu4", "control4"),alpha=0.05), na.rm=T)
ghuwtvsctrl04 <- na.omit(ghuwtvsctrl04, cols=ghumutvsctrl26$padj)
sum(ghuwtvsctrl04$padj < 0.05 & ghuwtvsctrl04$log2FoldChange > 1)
sum(ghuwtvsctrl04$padj < 0.05 & ghuwtvsctrl04$log2FoldChange < -1)
ggplot(data=ghuwtvsctrl04, aes(x=log2FoldChange, y=-log10(pvalue), col=ghuwtvsctrl04$diffexpressed, label=ghuwtvsctrl04$delabel)) + 
    geom_point() + 
    theme_minimal()
with(ghuwtvsctrl04, plot(log2FoldChange, -log10(pvalue), pch=20, main="wildtype GHu vs mock control (4dpi)", xlim=c(-10,15), ylim=c(0,10)))+
  with(subset(ghuwtvsctrl04, pvalue<.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="light blue"))+
  with(subset(ghuwtvsctrl04, abs(log2FoldChange)>1), points(log2FoldChange, -log10(pvalue), pch=20, col="gray"))+
  with(subset(ghuwtvsctrl04, pvalue<.05 & abs(log2FoldChange)>1), points(log2FoldChange, -log10(pvalue), pch=20, col="light green"))

ghuwtvsctrl17 <- data.frame(results(dds_out, contrast=c("group", "wildtype_GHu17", "control17"),alpha=0.05), na.rm=T)
ghuwtvsctrl17 <- na.omit(ghuwtvsctrl17, cols=ghuwtvsctrl17$padj)
sum(ghuwtvsctrl17$padj < 0.05 & ghuwtvsctrl17$log2FoldChange > 1)
sum(ghuwtvsctrl17$padj < 0.05 & ghuwtvsctrl17$log2FoldChange < -1)
ggplot(data=ghuwtvsctrl17, aes(x=log2FoldChange, y=-log10(pvalue), col=ghuwtvsctrl17$diffexpressed, label=ghuwtvsctrl17$delabel)) + 
    geom_point() + 
    theme_minimal()
with(ghuwtvsctrl17, plot(log2FoldChange, -log10(pvalue), pch=20, main="wildtype GHu vs mock control (17dpi)", xlim=c(-10,15), ylim=c(0,15)))+
  with(subset(ghuwtvsctrl17, pvalue<.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="light blue"))+
  with(subset(ghuwtvsctrl17, abs(log2FoldChange)>1), points(log2FoldChange, -log10(pvalue), pch=20, col="gray"))+
  with(subset(ghuwtvsctrl17, pvalue<.05 & abs(log2FoldChange)>1), points(log2FoldChange, -log10(pvalue), pch=20, col="light green"))

ghuwtvsctrl26 <- data.frame(results(dds_out, contrast=c("group", "wildtype_GHu26", "control26"),alpha=0.05), na.rm=T)
ghuwtvsctrl26 <- na.omit(ghuwtvsctrl26, cols=ghuwtvsctrl26$padj)
sum(ghuwtvsctrl26$padj < 0.05 & ghuwtvsctrl26$log2FoldChange > 1)
sum(ghuwtvsctrl26$padj < 0.05 & ghuwtvsctrl26$log2FoldChange < -1)
ggplot(data=ghuwtvsctrl26, aes(x=log2FoldChange, y=-log10(pvalue), col=ghuwtvsctrl26$diffexpressed, label=ghuwtvsctrl26$delabel)) + 
    geom_point() + 
    theme_minimal()
with(ghuwtvsctrl26, plot(log2FoldChange, -log10(pvalue), pch=20, main="wildtype GHu vs mock control (26dpi)", xlim=c(-10,15), ylim=c(0,10)))+
  with(subset(ghuwtvsctrl26, pvalue<.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="light blue"))+
  with(subset(ghuwtvsctrl26, abs(log2FoldChange)>1), points(log2FoldChange, -log10(pvalue), pch=20, col="gray"))+
  with(subset(ghuwtvsctrl26, pvalue<.05 & abs(log2FoldChange)>1), points(log2FoldChange, -log10(pvalue), pch=20, col="light green"))


```


```{r}

ghumutvsctrl04 <- data.frame(results(dds_out, contrast=c("group", "mutant_GHu4", "control4"),alpha=0.05), na.rm=T)
ghumutvsctrl04 <- na.omit(ghumutvsctrl04, cols=ghumutvsctrl26$padj)
sum(ghumutvsctrl04$padj < 0.05 & ghumutvsctrl04$log2FoldChange > 1)
sum(ghumutvsctrl04$padj < 0.05 & ghumutvsctrl04$log2FoldChange < -1)
ggplot(data=ghumutvsctrl04, aes(x=log2FoldChange, y=-log10(pvalue), col=ghumutvsctrl04$diffexpressed, label=ghumutvsctrl04$delabel)) + 
    geom_point() + 
    theme_minimal()
with(ghumutvsctrl04, plot(log2FoldChange, -log10(pvalue), pch=20, main="mutant GHu 1EK802G  vs mock control (4dpi)", xlim=c(-10,15), ylim=c(0,10)))+
  with(subset(ghumutvsctrl04, pvalue<.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="light blue"))+
  with(subset(ghumutvsctrl04, abs(log2FoldChange)>1), points(log2FoldChange, -log10(pvalue), pch=20, col="gray"))+
  with(subset(ghumutvsctrl04, pvalue<.05 & abs(log2FoldChange)>1), points(log2FoldChange, -log10(pvalue), pch=20, col="light green"))

ghumutvsctrl17 <- data.frame(results(dds_out, contrast=c("group", "mutant_GHu17", "control17"),alpha=0.05), na.rm=T)
ghumutvsctrl17 <- na.omit(ghumutvsctrl17, cols=ghumutvsctrl17$padj)
sum(ghumutvsctrl17$padj < 0.05 & ghumutvsctrl17$log2FoldChange > 1)
sum(ghumutvsctrl17$padj < 0.05 & ghumutvsctrl17$log2FoldChange < -1)
ggplot(data=ghumutvsctrl17, aes(x=log2FoldChange, y=-log10(pvalue), col=ghumutvsctrl17$diffexpressed, label=ghumutvsctrl17$delabel)) + 
    geom_point() + 
    theme_minimal()
with(ghumutvsctrl17, plot(log2FoldChange, -log10(pvalue), pch=20, main="mutant GHu 1EK802G vs mock control (17dpi)", xlim=c(-10,15), ylim=c(0,15)))+
  with(subset(ghumutvsctrl17, pvalue<.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="light blue"))+
  with(subset(ghumutvsctrl17, abs(log2FoldChange)>1), points(log2FoldChange, -log10(pvalue), pch=20, col="gray"))+
  with(subset(ghumutvsctrl17, pvalue<.05 & abs(log2FoldChange)>1), points(log2FoldChange, -log10(pvalue), pch=20, col="light green"))

ghumutvsctrl26 <- data.frame(results(dds_out, contrast=c("group", "mutant_GHu26", "control26"),alpha=0.05), na.rm=T)
ghumutvsctrl26 <- na.omit(ghumutvsctrl26, cols=ghumutvsctrl26$padj)
sum(ghumutvsctrl26$padj < 0.05 & ghumutvsctrl26$log2FoldChange > 1)
sum(ghumutvsctrl26$padj < 0.05 & ghumutvsctrl26$log2FoldChange < -1)
ggplot(data=ghumutvsctrl26, aes(x=log2FoldChange, y=-log10(pvalue), col=ghumutvsctrl26$diffexpressed, label=ghumutvsctrl26$delabel)) + 
    geom_point() + 
    theme_minimal()
with(ghumutvsctrl26, plot(log2FoldChange, -log10(pvalue), pch=20, main="mutant GHu 1EK802G vs mock control (26dpi)", xlim=c(-10,15), ylim=c(0,10)))+
  with(subset(ghumutvsctrl26, pvalue<.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="light blue"))+
  with(subset(ghumutvsctrl26, abs(log2FoldChange)>1), points(log2FoldChange, -log10(pvalue), pch=20, col="gray"))+
  with(subset(ghumutvsctrl26, pvalue<.05 & abs(log2FoldChange)>1), points(log2FoldChange, -log10(pvalue), pch=20, col="light green"))

```

```{r}
library(clusterProfiler)
library(gprofiler2)
#Gff2GeneTable("Niben1.gff")
read.gmt("GO_Niben3.gmt")
upload_GMT_file("output88.gmt")
read.gmt("output_file_gmt.gmt")
upload_GMT_file("GO_Niben3.gmt")
upload_GMT_file("output999.gmt")
# here is the custom ID : "gp__NEPK_ldZt_ZOw"
# new custom ID is: "gp__qbRo_UR6z_xEU"
upload_GMT_file("output9999.gmt")
gp__CkPw_qBVv_TQo

ghuwtvsmut4 <- data.frame(results(dds_out, contrast=c("group", "wildtype_GHu4", "mutant_GHu4"),alpha=0.05), na.rm=T)
ghuwtvsmut4_enrich <- noquote(rownames(subset(ghuwtvsmut4, pvalue<0.05 & abs(log2FoldChange)>1)))
lapply(ghuwtvsmut4_enrich, write, 'ghuwtmut4dpigenes.txt', append=T)
ghuwtvsmut4_result <-  gost(ghuwtvsmut17_enrich, organism = 'gp__CkPw_qBVv_TQo', significant = TRUE)
dataGO4 <- data.frame("condition" = "GHu WT vs GHu Mutant 4dpi", "GOghu" = ghuwtvsmut4_result$result$term_id,
"name" = ghuwtvsmut4_result$result$term_name, "ratio" = ghuwtvsmut4_result$result$term_size, "P.value" = ghuwtvsmut4_result$result$p_value)
ggplot(data = dataGO17, aes(x = "GHu WT vs GHu Mutant 4 dpi", y = name, 
                        color = P.value, size = ratio)) + 
  geom_point() +
  scale_color_gradient(low = "red", high = "blue") +
  theme_bw() + 
  ylab("") + 
  xlab("") + 
  ggtitle("GO enrichment analysis")


ghuwtvsmut17 <- data.frame(results(dds_out, contrast=c("group", "wildtype_GHu17", "mutant_GHu17"),alpha=0.05), na.rm=T)
ghuwtvsmut17_enrich <- noquote(rownames(subset(ghuwtvsmut17, pvalue<0.05 & abs(log2FoldChange)>1)))
lapply(ghuwtvsmut17_enrich, write, 'ghuwtmut17dpigenes.txt', append=T)
ghuwtvsmut17_result <-  gost(ghuwtvsmut17_enrich, organism = 'gp__CkPw_qBVv_TQo', significant = TRUE)
dataGO17 <- data.frame("condition" = "GHu WT vs GHu Mutant 17dpi", "GOghu" = ghuwtvsmut17_result$result$term_id,
"name" = ghuwtvsmut17_result$result$term_name, "ratio" = ghuwtvsmut17_result$result$term_size, "P.value" = ghuwtvsmut17_result$result$p_value)
ggplot(data = dataGO17, aes(x = "GHu WT vs GHu Mutant 17 dpi", y = name, 
                        color = P.value, size = ratio)) + 
  geom_point() +
  scale_color_gradient(low = "red", high = "blue") +
  theme_bw() + 
  ylab("") + 
  xlab("") + 
  ggtitle("GO enrichment analysis")

ghuwtvsmut26 <- data.frame(results(dds_out, contrast=c("group", "wildtype_GHu26", "mutant_GHu26"),alpha=0.05), na.rm=T)
ghuwtvsmut26_enrich <- noquote(rownames(subset(ghuwtvsmut26, pvalue<0.05 & abs(log2FoldChange)>1)))
lapply(ghuwtvsmut26_enrich, write, 'ghuwtmut26dpigenes.txt', append=T)
ghuwtvsmut26_result <-  gost(ghuwtvsmut26_enrich, organism = 'gp__CkPw_qBVv_TQo', significant = TRUE)
dataGO26 <- data.frame("condition" = "GHu WT vs GHu Mutant 26dpi", "GOghu" = ghuwtvsmut26_result$result$term_id,
"name" = ghuwtvsmut26_result$result$term_name, "ratio" = ghuwtvsmut26_result$result$term_size, "P.value" = ghuwtvsmut26_result$result$p_value)
ggplot(data = dataGO26, aes(x = "GHu WT vs GHu Mutant 26 dpi", y = name, 
                        color = P.value, size = ratio)) + 
  geom_point() +
  scale_color_gradient(low = "red", high = "blue") +
  theme_bw() + 
  ylab("") + 
  xlab("") + 
  ggtitle("GO enrichment analysis")





ridgeplot(ghuwtvsmut17_enrich)


```

```{r}
# ghuwtvsmut12_enrich <- noquote(rownames(subset(ghuwtvsmut12, padj<0.05 & abs(log2FoldChange)>1)))
# comparison of 12dpi does not return any gene set enrichment significance

f13wtvsmut4_enrich <- noquote(rownames(subset(f13wtvsmut4, padj<0.05 & abs(log2FoldChange)>1)))
f13wtvsmut4_enrich
f13wtvsmut4_result <-  gost(f13wtvsmut4_enrich, organism = 'gp__HTvx_rYJC_ffE', significant = TRUE)
dataGO4_F13 <- data.frame("condition" = "F13 WT vs F13 Mutant 4dpi","GOghu" = f13wtvsmut4_result$result$term_id,
"name" = f13wtvsmut4_result$result$term_name, "ratio" = f13wtvsmut4_result$result$term_size, "P.value" = f13wtvsmut4_result$result$p_value)

ggplot(data = dataGO4_F13, aes(x = "F13 WT vs F13 Mutant 4dpi", y = name, 
                        color = P.value, size = ratio)) + 
  geom_point() +
  scale_color_gradient(low = "red", high = "blue") +
  theme_bw() + 
  ylab("") + 
  xlab("") + 
  ggtitle("GO enrichment analysis")

ghuwtvsf13wt7_enrich <- noquote(rownames(subset(ghuwtvsf13wt7, padj<0.05 & abs(log2FoldChange)>1)))
ghuwtvsf13wt7_enrich
ghuwtvsf13wt7_result <-  gost(ghuwtvsf13wt7_enrich, organism = 'gp__HTvx_rYJC_ffE', significant = TRUE)
dataGO7_GHu_F13 <- data.frame("condition" = "GHu WT vs F13 WT 7dpi", "GOghu" = ghuwtvsf13wt7_result$result$term_id,
"name" = ghuwtvsf13wt7_result$result$term_name, "ratio" = ghuwtvsf13wt7_result$result$term_size, "P.value" = ghuwtvsf13wt7_result$result$p_value)

ggplot(data = dataGO7_GHu_F13, aes(x = "GHu WT vs F13 WT 7dpi", y = name, 
                        color = P.value, size = ratio)) + 
  geom_point() +
  scale_color_gradient(low = "red", high = "blue") +
  theme_bw() + 
  ylab("") + 
  xlab("") + 
  ggtitle("GO enrichment analysis")

# Overrepresentation of gene ontology for GHu WT and controls - returned insiginficant results for 4 and 12 dpi, but at 7dpi there were several ontologies found

#controlghu4 <- data.frame(results(dds_out, contrast=c("group", "ghu_wt4", "Control4"),alpha=0.05))
controlghu7 <- data.frame(results(dds_out, contrast=c("group", "ghu_wt7", "Control7"),alpha=0.05))
#controlghu12 <- data.frame(results(dds_out, contrast=c("group", "ghu_wt12", "Control12"),alpha=0.05))

#controlghu4_enrich <- noquote(rownames(subset(controlghu4, padj<0.05 & abs(log2FoldChange)>1)))
controlghu7_enrich <- noquote(rownames(subset(controlghu7, padj<0.05 & abs(log2FoldChange)>1)))
#controlghu12_enrich <- noquote(rownames(subset(controlghu12, padj<0.05 & abs(log2FoldChange)>1)))

#controlghu4_result <-  gost(controlghu4_enrich, organism = 'gp__3CN5_Ndmp_GXo', significant = TRUE)
controlghu7_result <-  gost(controlghu7_enrich, organism = 'gp__3CN5_Ndmp_GXo', significant = TRUE)
#controlghu12_result <-  gost(controlghu12_enrich, organism = 'gp__3CN5_Ndmp_GXo', significant = TRUE)

#dataGO4_controlghu <- data.frame("condition" = "Control vs GHu WT 4dpi", "GOghu" = controlghu4_result$result$term_id,
#"name" = controlghu4_result$result$term_name, "ratio" = controlghu4_result$result$term_size, "P.value" = controlghu4_result$result$p_value)
dataGO7_controlghu <- data.frame("condition" = "GHu WT vs Control 7dpi", "GOghu" = controlghu7_result$result$term_id,
"name" = controlghu7_result$result$term_name, "ratio" = controlghu7_result$result$term_size, "P.value" = controlghu7_result$result$p_value)
#dataGO12_controlghu <- data.frame("condition" = "Control vs GHu WT 12dpi", "GOghu" = controlghu12_result$result$term_id,
#"name" = controlghu12_result$result$term_name, "ratio" = controlghu12_result$result$term_size, "P.value" = # controlghu12_result$result$p_value)

ggplot(data = dataGO7_controlghu, aes(x = condition, y = name, 
                        color = P.value, size = ratio)) + 
  geom_point() +
  scale_color_gradient(low = "red", high = "blue") +
  theme_bw() + 
  ylab("") + 
  xlab("") + 
  ggtitle("") +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

controlf137 <- data.frame(results(dds_out, contrast=c("group", "f13_wt7", "Control7"),alpha=0.05))
controlf137_enrich <- noquote(rownames(subset(controlf137, padj<0.05 & abs(log2FoldChange)>1)))
controlf137_result <-  gost(controlf137_enrich, organism = 'gp__3CN5_Ndmp_GXo', significant = TRUE)
dataGO7_controlf13 <- data.frame("condition" = "F13 WT vs Control 7dpi", "GOghu" = controlf137_result$result$term_id,
"name" = controlf137_result$result$term_name, "ratio" = controlf137_result$result$term_size, "P.value" = controlf137_result$result$p_value)

ggplot(data = dataGO7_controlf13, aes(x = condition, y = name, 
                        color = P.value, size = ratio)) + 
  geom_point() +
  scale_color_gradient(low = "red", high = "blue") +
  theme_bw() + 
  ylab("") + 
  xlab("") + 
  ggtitle("") +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))
```

```{r}

exp_wide_ontology <- list(dataGO7_GHu_F13, dataGO7, dataGO4_F13, dataGO7_controlghu, dataGO7_controlf13)
library(reshape)
GO_plot_all <- Reduce(function(x,y) merge(x,y, all=TRUE), exp_wide_ontology, accumulate=FALSE)
GO_plot_all$name <- gsub("_", " ", GO_plot_all$name)

GO_plot_all$condition <- factor(GO_plot_all$condition, levels = c("F13 WT vs F13 Mutant 4dpi", "F13 WT vs Control 7dpi", "GHu WT vs Control 7dpi", "GHu WT vs F13 WT 7dpi", "GHu WT vs GHu Mutant 7dpi"))
ggplot(data = GO_plot_all, aes(x = condition, y = name, 
                        color = P.value, size = ratio)) + 
  geom_point() +
  scale_color_gradient(low = "red", high = "blue") +
  theme_bw() + 
  ylab("") + 
  xlab("") + 
  ggtitle("") +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), text = element_text(size=15))

# cleaning up a little

exp_wide_ontology <- list(dataGO7_GHu_F13, dataGO7, dataGO7_controlghu, dataGO7_controlf13)
library(reshape)
GO_plot_all <- Reduce(function(x,y) merge(x,y, all=TRUE), exp_wide_ontology, accumulate=FALSE)

GO_plot_all$condition <- factor(GO_plot_all$condition, levels = c("Control vs F13 WT 7dpi", "Control vs GHu WT 7dpi", "GHu WT vs F13 WT 7dpi", "GHu WT vs GHu Mutant 7dpi"))
ggplot(data = GO_plot_all, aes(x = condition, y = name, 
                        color = P.value, size = ratio)) + 
  geom_point() +
  scale_color_gradient(low = "red", high = "blue") +
  theme_bw() + 
  ylab("") + 
  xlab("") + 
  ggtitle("") +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), text = element_text(size=15))
```

```{r}
# String protein network analysis
# https://version-11-5.string-db.org/organism/STRG0071BOB 

BiocManager::install("STRINGdb")
library(STRINGdb)

string_db <- STRINGdb$new(version ="", species<-'STRG0071BOB', score_threshold=200, network_type="full")

#load annotation

#add_protein_description(screen)



```

```{r}
# NetSeekR https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-021-04554-1
```

```{r}
getGeneKEGGLinks()
getKEGGPathwayNames()

```

```{r}
#KEGG Annotation with BLASTKoala
#Kanehisa, M., Sato, Y., and Morishima, K.; BlastKOALA and GhostKOALA: KEGG tools for functional #characterization of genome and metagenome sequences. J. Mol. Biol. 428, 726-731 (2016).

# Submitted 2 tasks through the BLASTKOALA server, due to limitation of 50,000 genes per task and N. benthamiana genome size of >57,000 genes from Niben v1.0.1
# Files are saved as imported below
user_ko1.txt
user_ko2.txt
#merge tables


# Also tried using EGGNOG Annotation Mapper that took 16 minutes for entire N. benthamiana genome
# provides several output files, in gff, xlsx, and txt files
# import these files below to use clusterProfiler for enrichment analysis with not just GO terms, but also KO and KEGG, with bonus names

gene_annot <- read.delim("out.emapper.decorated.gff")

library(GenomicFeatures)
txdb <- makeTxDbFromGFF("out.emapper.decorated.gff")
tx<-transcriptsBy(txdb)



BiocManager::install("KEGGREST")
library(KEGGREST)

ghuwtvsmut04
ghuwtvsmut17
ghuwtvsmut26

ghuwtvsctrl04
ghuwtvsctrl17
ghuwtvsctrl26

ghumutvsctrl04
ghumutvsctrl17
ghumutvsctrl26



```




#This block is okay for a midpoint analysis with GO:XXXXXXX and followed by tab delineated geneID's
```{r}
# This generated output99.gmt and it looks great

# Install the 'readxl' package if it's not already installed
# install.packages("readxl")

# Load the 'readxl' package
library(readxl)

# Read the Excel file
excel_file <- read_excel("out.emapper.annotations1.xlsx")

# Create an empty list to store the GOs and corresponding GeneIDs
go_list <- list()

# Iterate over each row in the Excel file
for (i in 1:nrow(excel_file)) {
  # Get the GOs for the current row
  go_terms <- strsplit(as.character(excel_file$GOs[i]), ",")[[1]]
  
  # Get the GeneID for the current row
  gene_id <- excel_file$GeneID[i]
  
  # Iterate over each GO term
  for (go_term in go_terms) {
    # Check if the GO term already exists in the list
    if (go_term %in% names(go_list)) {
      # If it exists, append the GeneID to the existing entry
      go_list[[go_term]] <- paste(go_list[[go_term]], gene_id, sep = "\t")
    } else {
      # If it doesn't exist, create a new entry with the GeneID
      go_list[[go_term]] <- gene_id
    }
  }
}

# Write the GOs and corresponding GeneIDs to a .gmt file
gmt_file <- file("output99.gmt", "w")
for (go_term in names(go_list)) {
  gene_ids <- go_list[[go_term]]
  
  # Write the GO term and corresponding GeneIDs to the .gmt file
  cat(go_term, "\t", gene_ids, "\n", file = gmt_file)
}
close(gmt_file)


```


```{r}
# Install the 'readxl' and 'org.Hs.eg.db' packages if they're not already installed
# install.packages(c("readxl", "org.Hs.eg.db"))

# Load the required packages
library(readxl)
library(org.Hs.eg.db)

# Read the Excel file
excel_file <- read_excel("out.emapper.annotations1.xlsx")

# Create an empty list to store the GOs and corresponding GeneIDs
go_list <- list()

# Iterate over each row in the Excel file
for (i in 1:nrow(excel_file)) {
  # Get the GeneID for the current row
  gene_id <- excel_file$GeneID[i]
  
  # Retrieve the gene ontology terms for the GeneID
  go_terms <- AnnotationDbi::select(org.Hs.eg.db, keys = go_list, keytype = "GO", columns = "GO")$GO
  
  # Iterate over the retrieved GO terms
  for (go_term in go_terms) {
    # Check if the GO term already exists in the list
    if (go_term %in% names(go_list)) {
      # If it exists, append the GeneID to the existing entry
      go_list[[go_term]] <- paste(go_list[[go_term]], gene_id, sep = "\t")
    } else {
      # If it doesn't exist, create a new entry with the GeneID
      go_list[[go_term]] <- gene_id
    }
  }
}

# Write the GOs and corresponding GeneIDs to a .gmt file
gmt_file <- file("output101.gmt", "w")
for (go_term in names(go_list)) {
  gene_ids <- go_list[[go_term]]
  
  # Write the GO term, gene ontology, and corresponding GeneIDs to the .gmt file
  cat(go_term, "\t", "GO:", go_term, "\t", gene_ids, "\n", file = gmt_file)
}
close(gmt_file)

```

```{r}
# Install the 'readxl' and 'org.Hs.eg.db' packages if they're not already installed
# install.packages(c("readxl", "org.Hs.eg.db"))

# Load the required packages
library(readxl)
library(org.Hs.eg.db)

# Read the Excel file
excel_file <- read_excel("out.emapper.annotations1.xlsx")

# Create an empty list to store the GOs and corresponding GeneIDs
go_list <- list()

# Iterate over each row in the Excel file
for (i in 1:nrow(excel_file)) {
  # Get the GeneID for the current row
  gene_id <- excel_file$GeneID[i]
  
  # Get the GOs for the current row
  go_terms <- strsplit(as.character(excel_file$GOs[i]), ",")[[1]]
  
  # Iterate over each GO term
  for (go_term in go_terms) {
    # Check if the GO term already exists in the list
    if (go_term %in% names(go_list)) {
      # If it exists, append the GeneID to the existing entry
      go_list[[go_term]] <- paste(go_list[[go_term]], gene_id, sep = "\t")
    } else {
      # If it doesn't exist, create a new entry with the GeneID
      go_list[[go_term]] <- gene_id
    }
  }
}

# Retrieve gene ontology terms for the keys in go_list
go_terms <- keys(org.Hs.eg.db, keytype = "GO", key = names(go_list))

# Write the GOs and corresponding GeneIDs to a .gmt file
gmt_file <- file("output102.gmt", "w")
for (i in 1:length(go_terms)) {
  go_term <- go_terms[i]
  gene_ids <- go_list[[go_term]]
  
  # Write the GO term, gene ontology, and corresponding GeneIDs to the .gmt file
  cat(go_term, "\t", "GO:", go_term, "\t", gene_ids, "\n", file = gmt_file)
}
close(gmt_file)

```

```{r}
# Install the 'readxl' and 'org.Hs.eg.db' packages if they're not already installed
# install.packages(c("readxl", "org.Hs.eg.db"))

# Load the required packages
library(readxl)
library(org.Hs.eg.db)
library(GO.db)

# Read the Excel file
excel_file <- read_excel("out.emapper.annotations1.xlsx")

# Create an empty list to store the GOs and corresponding GeneIDs
go_list <- list()

# Iterate over each row in the Excel file
for (i in 1:nrow(excel_file)) {
  # Get the GeneID for the current row
  gene_id <- excel_file$GeneID[i]
  
  # Get the GOs for the current row
  go_terms <- strsplit(as.character(excel_file$GOs[i]), ",")[[1]]
  
  # Iterate over each GO term
  for (go_term in go_terms) {
    # Check if the GO term already exists in the list
    if (go_term %in% names(go_list)) {
      # If it exists, append the GeneID to the existing entry
      go_list[[go_term]] <- paste(go_list[[go_term]], gene_id, sep = "\t")
    } else {
      # If it doesn't exist, create a new entry with the GeneID
      go_list[[go_term]] <- gene_id
    }
  }
}

go_names <- Term(GO.db, keys = names(go_list), columns = "TERM")

# Write the GOs and corresponding GeneIDs to a .gmt file
gmt_file <- file("path/to/your/new/file.gmt", "w")
for (i in 1:length(go_names)) {
  go_term <- names(go_list)[i]
  go_name <- go_names[[go_term]]
  gene_ids <- go_list[[go_term]]
  
  # Write the GO term, gene ontology term name, and corresponding GeneIDs to the .gmt file
  cat(go_term, "\t", go_name, "\t", gene_ids, "\n", file = gmt_file)
}
close(gmt_file)

```



```{r}

# Install the 'readxl' and 'GO.db' packages if they're not already installed
# install.packages(c("readxl", "GO.db"))

# Load the required packages
library(readxl)
library(GO.db)

# Read the Excel file
excel_file <- read_excel("out.emapper.annotations1.xlsx")

# Create an empty list to store the GOs and corresponding GeneIDs
go_list <- list()

# Iterate over each row in the Excel file
for (i in 1:nrow(excel_file)) {
  # Get the GeneID for the current row
  gene_id <- excel_file$GeneID[i]
  
  # Get the GOs for the current row
  go_terms <- strsplit(as.character(excel_file$GOs[i]), ",")[[1]]
  
  # Iterate over each GO term
  for (go_term in go_terms) {
    # Check if the GO term already exists in the list
    if (go_term %in% names(go_list)) {
      # If it exists, append the GeneID to the existing entry
      go_list[[go_term]] <- paste(go_list[[go_term]], gene_id, sep = "\t")
    } else {
      # If it doesn't exist, create a new entry with the GeneID
      go_list[[go_term]] <- gene_id
    }
  }
}

# Retrieve gene ontology term names for the keys in go_list
go_names <- Term(GO.db, keys = names(go_list), columns = "term_name")

# Write the GOs and corresponding GeneIDs to a .gmt file
gmt_file <- file("output33.gmt", "w")
for (i in 1:length(go_names)) {
  go_term <- names(go_list)[i]
  go_name <- go_names[[go_term]]
  gene_ids <- go_list[[go_term]]
  
  # Write the GO term, gene ontology term name, and corresponding GeneIDs to the .gmt file
  cat(go_term, "\t", go_name, "\t", gene_ids, "\n", file = gmt_file)
}
close(gmt_file)


```

```{r}
library(gprofiler2)

# Example gene ontology ID numbers
goid_list <- c("GO:0008150", "GO:0005575", "GO:0003674")

# Convert gene ontology ID numbers to human-readable names
name_list <- gprofiler(goid_list, organism = "hsapiens", key = NULL)$info[, "name"]

# Print the results
for (i in seq_along(goid_list)) {
  cat(goid_list[i], ": ", name_list[i], "\n")
}
```

```{r}

library(readxl)
library(gprofiler2)
library(openxlsx)

# Load the Excel file
excel_file <- "path/to/your/file.xlsx"  # Replace with the path to your Excel file
df <- read_excel(excel_file)

# Extract the gene ontology ID numbers from the first column
goid_list <- df[[1]]

# Convert gene ontology ID numbers to human-readable names
name_list <- gprofiler(goid_list, organism = "hsapiens", key = NULL)$info[, "name"]

# Add a new column with the human-readable names
df$Human_Readable_Names <- name_list

# Save the updated data frame to a new Excel file
output_file <- "path/to/your/output/file.xlsx"  # Replace with the path for your output file
write.xlsx(df, output_file, row.names = FALSE)


```


```{r}
library(stringr)

# Read the file
data <- readLines("Niben101_annotation.functional.txt")

# Create an empty data frame to store the results
result <- data.frame(GeneOntologyCode = character(),
                     GeneOntologyName = character(),
                     stringsAsFactors = FALSE)

# Loop through each row in the input data
for (row in data) {
  # Extract gene ontology codes and names using regular expressions
  ontology_codes <- str_extract_all(row, "GO:\\d+")
  ontology_names <- str_extract_all(row, "\\([^)]+\\)")
  
  # Check if there are any matches before creating the row data frame
  if (length(ontology_codes[[1]]) > 0 && length(ontology_names[[1]]) > 0) {
    # Flatten the lists of codes and names into character vectors
    ontology_codes <- unlist(ontology_codes)
    ontology_names <- unlist(ontology_names)
    
    # Remove leading/trailing spaces and parentheses from the names
    ontology_names <- str_replace_all(ontology_names, "[\\(\\)]", "")
    
    # Create a data frame with each combination of code and name
    row_data <- data.frame(GeneOntologyCode = ontology_codes,
                           GeneOntologyName = ontology_names,
                           stringsAsFactors = FALSE)
    
    # Append the row data to the result data frame
    result <- rbind(result, row_data)
  }
}

# Remove duplicates from the result data frame
result <- unique(result)

# Export the result to a new text file
write.table(result, "output.annot.txt", sep = "\t", row.names = FALSE, quote = FALSE)


```

```{r}

# Read the file
data <- readLines("NibenGO.txt")

# Create an empty data frame to store the results
result <- data.frame(GeneOntologyCode = character(),
                     GeneOntologyDescription = character(),
                     stringsAsFactors = FALSE)

# Loop through each line in the file
for (line in data) {
  # Extract gene ontology code and description
  ontology_code <- str_match(line, "(GO:\\d+)\\s")[, 2]
  ontology_description <- str_match(line, "\\((.*?)\\)")[, 2]
  
  # Check if the code and description are not empty
  if (!is.na(ontology_code) && !is.na(ontology_description)) {
    # Create a data frame with the code and description
    row_data <- data.frame(GeneOntologyCode = ontology_code,
                           GeneOntologyDescription = ontology_description,
                           stringsAsFactors = FALSE)
    
    # Append the row data to the result data frame
    result <- rbind(result, row_data)
  }
}

# Remove duplicates from the result data frame
result <- unique(result)

# Export the result to a new text file
write.table(result, "output_GO1.txt", sep = "\t", row.names = FALSE, quote = FALSE)

```

```{r}

# Read the two input files
file1 <- readLines("output_GO1.txt")
file2 <- readLines("output99.txt")

# Create an empty data frame to store the results
result <- data.frame(GeneOntologyCode = character(),
                     GeneOntologyDescription = character(),
                     GeneIDs = character(),
                     stringsAsFactors = FALSE)

# Loop through each line in file1
for (line in file1) {
  # Extract gene ontology code and description
  ontology_code <- str_match(line, "(GO:\\d+)")[, 2]
  ontology_description <- str_match(line, "(?<=\t).*$")[, 1]
  
  # Check if the code and description are not empty
  if (!is.na(ontology_code) && !is.na(ontology_description)) {
    # Match the GeneIDs from file2 based on the GeneOntologyCode
    matching_ids <- file2[grep(ontology_code, file2)]
    
    # Create a data frame with the description and matching GeneIDs
    row_data <- data.frame(GeneOntologyDescription = ontology_description,
                           GeneIDs = paste(matching_ids, collapse = "\t"),
                           stringsAsFactors = FALSE)
    
    # Append the row data to the result data frame
    result <- rbind(result, row_data)
  }
}

# Remove duplicates from the result data frame
result2 <- unique(result)
result3 <- result2[, c(2,1,3)]
df <- df[, c(2, 1, 3)]

# Export the result to a new text file
write.table(result, "finalNiben101GO.gmt", sep = "\t", row.names = FALSE, quote = FALSE)



```

```{r}
# Load required packages
BiocManager::install("org.At.tair.db")
library(org.At.tair.db)

# Read the file as lines
lines <- readLines("output999.txt")

# Split each line by the tab character
data <- strsplit(lines, "\t")

# Extract the first column (GO Number)
go_numbers <- sapply(data, function(x) x[[1]])

# Get unique GO numbers
unique_go_numbers <- unique(go_numbers)

# Get gene ontology descriptions
descriptions <- unlist(AnnotationDbi::select(org.At.tair.db, keys = unique_go_numbers, keytype = "GO", columns = "PFAM"))

# Replace "NA" values with gene ontology descriptions
for (i in seq_along(data)) {
  if (data[[i]][[2]] == "NA") {
    data[[i]][[2]] <- descriptions[go_numbers[i] == unique_go_numbers]
  }
}

# Reconstruct lines with updated data
updated_lines <- sapply(data, function(x) paste(x, collapse = "\t"))

# Save the updated data
writeLines(updated_lines, "output999_updated.txt")



```
```{r}
# Load required libraries
library(httr)
library(jsonlite)

# Read the file
data <- read.table("output999.txt", sep = "\t", header = FALSE, stringsAsFactors = FALSE, fill=TRUE)

# Extract unique GOIDs from the first column
goids <- unique(data$V1)

# Create a function to retrieve GO term description from the API
getGoTerm <- function(goid) {
  url <- paste0("http://api.geneontology.org/api/ontology/term/", goid, "/")
  response <- GET(url)
  content <- fromJSON(rawToChar(response$content))
  term <- content$label
  return(term)
}

# Iterate over each GOID, retrieve the term, and replace "NA" with the term
for (i in seq_along(goids)) {
  goid <- goids[i]
  term <- getGoTerm(goid)
  data[data$V1 == goid, -c(1, 2)] <- term
}

# Write the updated data to a new file
write.table(data, "updated_output999.txt", sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)

# Load required libraries
library(httr)
library(jsonlite)

# Read the file
data <- read.table("output888.txt", sep = "\t", header = FALSE, stringsAsFactors = FALSE, fill = TRUE)

# Extract unique GOIDs from the first column
goids <- unique(data$V1)

# Create a function to retrieve GO term description from the API
getGoTerm <- function(goid) {
  url <- paste0("http://api.geneontology.org/api/ontology/term/", goid, "/")
  response <- GET(url)
  content <- fromJSON(rawToChar(response$content))
  term <- content$label
  return(term)
}

# Iterate over each GOID, retrieve the term, and replace the second column with the term
for (i in seq_along(goids)) {
  goid <- goids[i]
  term <- getGoTerm(goid)
  data[data$V1 == goid,][, 2] <- term
  # Print progress every 200 iterations
  if (i %% 500 == 0) {
    cat("Processed", i, "out of", length(goids), "queries\n")
  }
}
#If an error arises here, you may have to initiate the second column in the dataframe 'data'



# Write the updated data to a new file
write.table(data, "updated_output999.txt", sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)






```


```{r}

# Read the input file
input_file <- "Niben101_annotation.functional.txt"
file_lines <- readLines(input_file)

# Initialize an empty vector to store the parsed data
go_numbers <- c()

# Parse the file lines
for (line in file_lines) {
  # Extract all GO numbers followed by closing parenthesis
  matches <- gregexpr("GO:\\d+\\s*\\([^)]+\\)", line, perl = TRUE)
  if (length(matches[[1]]) > 0) {
    go_matches <- regmatches(line, matches)[[1]]
    
    # Append to the vector
    go_numbers <- c(go_numbers, go_matches)
  }
}

# Count the number of unique entries for each GO number
count <- table(go_numbers)

# Print the counts
for (i in seq_along(count)) {
  cat(paste("GO:", names(count)[i], " - ", count[i], " unique entries\n"))
}




# Install the readxl package if not already installed
# install.packages("readxl")

# Load the readxl package
library(readxl)

# Read the Excel file
data <- read_excel("out.emapper.annotations1.xlsx", sheet = 1)

# Extract the second column
column_data <- data[, 2]

# Initialize a vector to store unique GO IDs
go_ids <- character()

# Process each entry in the second column
for (entry in column_data) {
  # Split the entry by commas
  entries <- unlist(strsplit(as.character(entry), ","))

  # Process each entry
  for (entry in entries) {
    # Check if it starts with 'GO:' (indicating a GO ID)
    if (grepl("^GO:", entry)) {
      # Add the GO ID to the vector
      go_ids <- c(go_ids, entry)
    }
  }
}

# Get the total number of unique GO IDs
total_go_ids <- length(unique(go_ids))

# Print the result
cat("Total unique GO IDs:", total_go_ids, "\n")




```


