---
title: "WGCNA Analysis"
author: "Brandon Roy"
date: '2022-08-16'
output: html_document
---

```{r}
BiocManager::install("WGCNA")
install.packages(c("matrixStats", "Hmisc", "splines", "foreach", "doParallel", "fastcluster", "dynamicTreeCut", "survival", "BiocManager"))
BiocManager::install(c("GO.db", "preprocessCore", "impute"))
library(WGCNA)
library(DESeq2)
set.seed(2)

```

```{r}
#Visualization of MEs on timepoint
#summary of number of gene per module
dds_interaction <- DESeqDataSetFromMatrix(countData = cts, colData = coldata, design = ~ group)
dds_out_interaction <- DESeq(dds_interaction)
vsd_interaction <- vst(dds_out_interaction,blind=FALSE)
cts_vst_i <- assay(vsd_interaction)
cts_vst_i <- as.data.frame(cts_vst_i)

options(stringsAsFactors = FALSE)
datExpr0 = as.data.frame(t(cts_vst_i))
datExpr0 <- as.matrix(datExpr0)



#Chooseasetofsoft-thresholdingpowers 
powers=c(c(1:10),seq(from=12,to=20,by=2)) 
#Callthenetworktopologyanalysisfunction 
sft=pickSoftThreshold(datExpr0,powerVector=powers,verbose=5) 
#Plottheresults: 
sizeGrWindow(5,5) 
par(mfrow=c(1,2)); 
cex1=0.9; 
#Scale-freetopologyfitindexasafunctionofthesoft-thresholdingpower 
plot(sft$fitIndices[,1],-sign(sft$fitIndices[,3])*sft$fitIndices[,2], 
     xlab="SoftThreshold(power)",ylab="ScaleFreeTopologyModelFit,signedR^2",type="n", 
     main=paste("Scaleindependence")) + 
  text(sft$fitIndices[,1],-sign(sft$fitIndices[,3])*sft$fitIndices[,2], labels=powers,cex=cex1,col="red") +
  abline(h=0.85,col="red") 
#thislinecorrespondstousinganR^2cut-offofh 

# use 5 as threshold at 0.85

#Meanconnectivityasafunctionofthesoft-thresholdingpower 
plot(sft$fitIndices[,1],sft$fitIndices[,5], 
     xlab="SoftThreshold(power)",ylab="MeanConnectivity",type="n", 
     main=paste("Meanconnectivity")) + 
  text(sft$fitIndices[,1],sft$fitIndices[,5],labels=powers,cex=cex1,col="red")





```




```{r}
# Clustering using eigenmodules into groups with at least 200 genes per module
library(WGCNA)
net = blockwiseModules(datExpr0_root, maxBlockSize = 20000, 
                       power=5, networkType = "unsigned", TOMType = "signed", 
                       minModuleSize = 10, reassignThreshold = 0, mergeCutHeight = 0.15,
                       minKMEtoStay = 0.3, numericLabels = TRUE, pamRespectsDendro = FALSE, saveTOMs = TRUE,
                       saveTOMFileBase = "supervised_norm_count_TOM3", verbose = 3)
# REMEMBER TO RESTART SESSION AND RERUN LIBRARY(WGCNA) ONLY TO AVOID ERRORS

table(net$colors)
mergedColors = labels2colors(net$colors)
table(mergedColors)

moduleLabels = net$colors

moduleColors = labels2colors(net$colors)
MEs = net$MEs;
geneTree = net$dendrograms[[1]];
save(MEs, moduleLabels, moduleColors, geneTree,
     file = "unsupervise-03-networkConstruction-auto_n50=.RData")

#Check eigengenes
MEs0 = moduleEigengenes(datExpr0, moduleColors)$eigengenes
MEs = orderMEs(MEs0)
modNames = substring(names(MEs), 3)

nGenes = ncol(datExpr0)
nSamples = nrow(datExpr0)
moduleTraitCor = cor(MEs, coldata[,c(2,3)], use = "p")
moduleTraitPvalue = corPvalueStudent(moduleTraitCor, nSamples)
moduleTraitCor
View(moduleTraitPvalue)
textMatrix = paste(signif(moduleTraitCor, 2), "\n(",
                   signif(moduleTraitPvalue, 1), ")", sep = "");
dim(textMatrix) = dim(moduleTraitCor)
par(mar = c(4,6, 4, 1))

table(mergedColors)
N_gene_per_module <- as.data.frame(table(mergedColors))
names(N_gene_per_module) = c("module_name", "Gene_number")
N_gene_per_module <- N_gene_per_module[order(-N_gene_per_module$Gene_number),]
module_name <- N_gene_per_module$module_name
N_gene_per_module$name_and_number <- paste("ME",N_gene_per_module$module_name," (",N_gene_per_module$Gene_number,")", sep ="")


#Relabelblockwisemodules 
bwLabels=matchLabels(net$colors,moduleLabels)
#Convertlabelstocolorsforplotting 
bwModuleColors=labels2colors(bwLabels)

table(bwLabels)
#Plotthedendrogramandthemodulecolorsunderneathforblock1 
plotDendroAndColors(net$dendrograms[[1]],bwModuleColors[net$blockGenes[[1]]], "Modulecolors",main="Genedendrogramandmodulecolorsinblock1", dendroLabels=FALSE,hang=0.03, addGuide=TRUE,guideHang=0.05) 
#Plotthedendrogramandthemodulecolorsunderneathforblock2 
plotDendroAndColors(net$dendrograms[[10]],bwModuleColors[net$blockGenes[[10]]], "Modulecolors",main="Genedendrogramandmodulecolorsinblock2", dendroLabels=FALSE,hang=0.03, addGuide=TRUE,guideHang=0.05)

plotDendroAndColors(geneTree,
                    cbind(net$moduleColors, bwModuleColors),
                    c("Single block", "2 blocks"),
                    dendroLabels = F, hang = 0.03,
                    addGuide = T, guideHang = 0.05)




library(dplyr)
library(tidyr)
library(ggiraphExtra)
library(ggplot2)
```

```{r}
# Model eigengene visualization for treatment
sample_name <- row.names(coldata)
MEs_visualization <- data.frame(sample_name,MEs,coldata)
MEs_visualization$Treatment <-factor(MEs_visualization$Treatment)

MEs_visualization_long <- gather(MEs_visualization,Module,ME,MEblack:MEgrey, factor_key=TRUE)
MEs_visualization_long <- gather(MEs_visualization,Module,ME,MEturquoise:MEorangered3, factor_key=TRUE)

#Summarizing
MEs_visualization_long_summary <- summarySE(MEs_visualization_long, measurevar="ME", groupvars=c("Module","Time","Treatment"))
MEs_visualization_long_summary_new <- MEs_visualization_long_summary
pd <- position_dodge(0.2)
module_levels = paste0("ME",N_gene_per_module$module_name)
MEs_visualization_long_summary_new$Module <- factor(MEs_visualization_long_summary_new$Module,levels = module_levels,labels  = N_gene_per_module$name_and_number)
MEs_visualization_long_summary_new$Treatment <- factor(MEs_visualization_long_summary_new$Treatment)

MEs_visualization_1 <- ggplot(MEs_visualization_long_summary_new,aes(x=Time, y=ME, group = Treatment, color = Treatment)) +
  geom_line(position=pd) +
  geom_point(position=pd, size = 1.7) +
  facet_wrap( ~ Module, scales = "free_y",ncol = 5) + 
  xlab("Days post inoculation") +
  ylab("Module eigengene") +
  theme_bw() + 
  theme(axis.line = element_line(colour = "black")) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  theme(legend.direction = "vertical", legend.box = "vertical") + 
  theme(axis.text = element_text(color = "black",size = 10)) + 
  theme(strip.text=element_text(color="Black", size = 10, face = "bold"), strip.background = element_blank()) +
  theme(axis.title = element_text(color = "black",size = 14, face = "bold" )) +
  labs(color = "Treatment") +
  scale_color_manual(values = c("gray","orange","#d1495b"), labels= c('Control','GHu Mutant','GHu WT')) +
  theme(axis.text.y = element_blank(),axis.ticks.y =  element_blank()) +
  theme(legend.text = element_text(color="Black", size = 12, face = "bold")) +
  theme(legend.title = element_text(color="Black", size = 12, face = "bold")) +
  theme(legend.position = 'top', legend.direction = 'horizontal')

par(mar = c(3,10, 3, 0))
MEs_visualization_1 

gtable_show_names(MEs_visualization_1)


MEs_visualization_1 <- reposition_legend(MEs_visualization_1 + facet_wrap(~Module, ncol=7,scales = "free_y"), 'center', panel='caption')

ggsave("MEs_visualization_figure2.png", plot = MEs_visualization_1, width = 35, height = 50, unit = "cm", dpi = 1000)

ggplot(MEs_visualization_1$data$Module)
```


```{r}
# Model eigengene visualization for time
library(tidyr)
library(Rmisc)
sample_name <- row.names(coldata)
MEs_visualization <- data.frame(sample_name,MEs,coldata)
MEs_visualization
MEs_visualization$Time <-factor(MEs_visualization$Time)

MEs_visualization_long <- gather(MEs_visualization,Module,ME,MEred:MEgrey, factor_key=TRUE)

#Summarizing

MEs_visualization_long_summary <- summarySE(MEs_visualization_long, measurevar="ME", groupvars=c("Module","Time","Treatment"))
MEs_visualization_long_summary_new <- MEs_visualization_long_summary
pd <- position_dodge(0.2)
module_levels = paste0("ME",N_gene_per_module$module_name)
MEs_visualization_long_summary_new$Module <- factor(MEs_visualization_long_summary_new$Module,levels = module_levels,labels  = N_gene_per_module$name_and_number)
MEs_visualization_long_summary_new$Time <- factor(MEs_visualization_long_summary_new$Time)


MEs_visualization_long_summary_new$Time <- as.factor(MEs_visualization_long_summary_new$Time)
ggplot(MEs_visualization_long_summary_new, aes(x=Time, y=ME, group = Time, color = Treatment)) +
  geom_line(position=pd) +
  geom_point(position=pd, size = 1.7) +
  facet_wrap( ~ Module,scales = "free_y",ncol = 9) + 
  xlab("Days post inoculation") +
  ylab("Module eigengene") +
  theme_bw() + 
  theme(axis.line = element_line(colour = "black")) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  theme(legend.direction = "vertical", legend.box = "vertical") + 
  theme(axis.text = element_text(color = "black",size = 10)) + 
  theme(strip.text=element_text(color="Black", size = 10, face = "bold"), strip.background = element_blank()) +
  theme(axis.title = element_text(color = "black",size = 14, face = "bold" )) +
  labs(color = "Treatment") +
  scale_color_manual(values = c("grey","red","orange")) +
  theme(axis.text.y = element_blank(),axis.ticks.y =  element_blank()) +
  theme(legend.text = element_text(color="Black", size = 12, face = "bold")) +
  theme(legend.title = element_text(color="Black", size = 12, face = "bold")) +
  scale_x_continuous(breaks = c(4,17,26))

MEs_visualization_1 
gtable_show_names(MEs_visualization_1)


MEs_visualization_1 <- reposition_legend(MEs_visualization_1 + facet_wrap(~Module, ncol=4,scales = "free_y"),
                                                     'center', panel=c('panel-4-3'))

ggsave("MEs_visualization_figure.png", plot = MEs_visualization_1, width = 16, height = 15, unit = "cm", dpi = 1500)

```

```{r}
# find the best WGCNA results
BiocManager::install("limma")
library(limma)
module_eigengenes <- net$MEs
#Ensure that data tables are the same and align
all.equal(rownames(coldata), rownames(module_eigengenes))
coldata$refinebio_accession_code <- rownames(coldata)
des_mat <- model.matrix(~ coldata$Treatment)
# lmFit() needs a transposed version of the matrix
fit <- limma::lmFit(t(module_eigengenes), design = des_mat)
# Apply empirical Bayes to smooth standard errors
fit <- limma::eBayes(fit)
# Apply multiple testing correction and obtain stats
stats_df <- limma::topTable(fit, number = ncol(module_eigengenes)) %>%
  tibble::rownames_to_column("module")
head(stats_df)

# The most differentially expressed modules appear to be MEs 7,3,0,1,13,12,2, & 5
# 13, 1, 6, 9 , 11, 12 for Roots

gene_module_key <- tibble::enframe(net$colors, name = "gene", value = "module") %>%
  # Let's add the `ME` part so its more clear what these numbers are and it matches elsewhere
  dplyr::mutate(module = paste0("ME", module))


module_7_df <- module_eigengenes %>%
  tibble::rownames_to_column("accession_code") %>%
  # Here we are performing an inner join with a subset of metadata
  dplyr::inner_join(coldata %>%
    dplyr::select(refinebio_accession_code, group, Treatment, Time),
  by = c("accession_code" = "refinebio_accession_code")
  )


ggplot(module_7_df, aes(x = Treatment, y = ME26, color = Treatment)) +
  geom_boxplot(width = 0.2, outlier.shape = NA) +
  ggforce::geom_sina(maxwidth = 0.3)  +
  facet_wrap("Time")
ggplot(module_7_df, aes(x = Treatment, y = ME8, color = Treatment)) +
  geom_boxplot(width = 0.2, outlier.shape = NA) +
  ggforce::geom_sina(maxwidth = 0.3)  +
  facet_wrap("Time")  
ggplot(module_7_df, aes(x = Treatment, y = ME47, color = Treatment)) +
  geom_boxplot(width = 0.2, outlier.shape = NA) +
  ggforce::geom_sina(maxwidth = 0.3)  +
  facet_wrap("Time")  
ggplot(module_7_df, aes(x = Treatment, y = ME20, color = Treatment)) +
  geom_boxplot(width = 0.2, outlier.shape = NA) +
  ggforce::geom_sina(maxwidth = 0.3)  +
  facet_wrap("Time")
ggplot(module_7_df, aes(x = Treatment, y = ME39, color = Treatment)) +
  geom_boxplot(width = 0.2, outlier.shape = NA) +
  ggforce::geom_sina(maxwidth = 0.3)  +
  facet_wrap("Time")
ggplot(module_7_df, aes(x = Treatment, y = ME19, color = Treatment)) +
  geom_boxplot(width = 0.2, outlier.shape = NA) +
  ggforce::geom_sina(maxwidth = 0.3)  +
  facet_wrap("Time")  
ggplot(module_7_df, aes(x = Treatment, y = ME35, color = Treatment)) +
  geom_boxplot(width = 0.2, outlier.shape = NA) +
  ggforce::geom_sina(maxwidth = 0.3)  +
  facet_wrap("Time")  
ggplot(module_7_df, aes(x = Treatment, y = ME9, color = Treatment)) +
  geom_boxplot(width = 0.2, outlier.shape = NA) +
  ggforce::geom_sina(maxwidth = 0.3)  +
  facet_wrap("Time")


```

```{r}
library(gprofiler2)
#  Genes down regulated at 26 dpi only for wildtype GHu
ME2_genes <- gene_module_key %>%
  dplyr::filter(module == "ME1")
gp__3CN5_Ndmp_GXo
GHu_WT_genes <- gene_module_key %>%
  dplyr::filter(module== "ME39")
ghugores <-  gost(GHu_WT_genes$gene, organism = 'gp__CkPw_qBVv_TQo', significant = F)
ghugores$result
dataghu7GO <- data.frame("Enrichment" = "ME wildtype 26dpi", "GOghu" = ghugores$result$term_id,
"name" = ghugores$result$term_name, "ratio" = ghugores$result$term_size, "P.value" = ghugores$result$p_value)

ggplot(data = dataghu7GO, aes(x = "ME GHu 26dpi", y = name,
                        color = P.value, size = ratio)) + 
  geom_point() +
  scale_color_gradient(low = "red", high = "blue") +
  theme_bw() + 
  ylab("") + 
  xlab("") + 
  ggtitle("GO enrichment analysis")
dataghu7GO$name <- gsub("_", " ", dataghu7GO$name)
#dataghu7GO <- dataghu7GO %>%
 # dplyr::filter(P.value <0.045)
dataghu7GO <- dataghu7GO %>%
  dplyr::filter(P.value <0.25)
ggplot(dataghu7GO, aes(x = reorder(name,ratio), y = ratio, fill = P.value)) +
  geom_bar(stat = "identity", position = "dodge", ) +
  scale_fill_gradient(low = "red", high = "blue") +
  coord_flip() + 
  xlab("Gene ontology term") + 
  ylab("Ratio of GO terms present") +
  guides(fill=guide_legend(title="P value")) +
  theme(
    legend.position = c(.975, .05),
    legend.justification = c("right", "bottom"),
    legend.box.just = "top",
    legend.margin = margin(6, 6, 6, 6),
    plot.margin = unit(c(0,5,0,0),"cm")
    )
```

```{r}
F13_mut4_genes <- gene_module_key %>%
  dplyr::filter(module == "ME27" | module == "ME11" | module == "ME9")
f13mut4res <-  gost(F13_mut4_genes$gene, organism = 'gp__3CN5_Ndmp_GXo', significant = TRUE)
dataf13m4GO <- data.frame("GOghu" = f13mut4res$result$term_id,
"name" = f13mut4res$result$term_name, "ratio" = f13mut4res$result$term_size, "P.value" = f13mut4res$result$p_value)
dataf13m4GO$name <- gsub("_", " ", dataf13m4GO$name)
ggplot(dataf13m4GO, aes(x = reorder(name,ratio), y = ratio, fill = P.value)) +
  geom_bar(stat = "identity", position = "dodge", ) +
  scale_fill_gradient(low = "red", high = "blue") +
  coord_flip() + 
  xlab("Gene ontology term") + 
  ylab("Ratio of GO terms present") +
  guides(fill=guide_legend(title="P value")) +
  theme(
    legend.position = c(.975, .05),
    legend.justification = c("right", "bottom"),
    legend.box.just = "right",
    legend.margin = margin(6, 6, 6, 6))
```


```{r}
# no significant resulting GO's for f13mut 7dpi
F13_mut7_genes <- gene_module_key %>%
  dplyr::filter(module == "ME23" | module == "ME28")
f13mut7res <-  gost(F13_mut7_genes$gene, organism = 'gp__HTvx_rYJC_ffE', significant = FALSE)

# no significant resulting GO's for f13mut 12dpi
F13_mut12_genes <- gene_module_key %>%
  dplyr::filter(module == "ME25" | module == "ME26")
f13mut12res <-  gost(F13_mut12_genes$gene, organism = 'gp__HTvx_rYJC_ffE', significant = TRUE)
```

```{r}
# no significant resulting GO's for GHu mutant 4dpi
ghu_mut4_genes <- gene_module_key %>%
  dplyr::filter(module == "ME14")
ghumut4res <-  gost(ghu_mut4_genes$gene, organism = 'gp__HTvx_rYJC_ffE', significant = TRUE)
# no significant resulting GO's for GHu mutant 7dpi
ghu_mut7_genes <- gene_module_key %>%
  dplyr::filter(module == "ME22" | module == "ME21")
ghumut7res <-  gost(ghu_mut7_genes$gene, organism = 'gp__HTvx_rYJC_ffE', significant = FALSE)
```

```{r}
ME2gores <-  gost(ME2_genes$gene, organism = 'gp__HTvx_rYJC_ffE', significant = TRUE)
dataME2GO <- data.frame("GOghu" = ME2gores$result$term_id,
"name" = ME2gores$result$term_name, "ratio" = ME2gores$result$term_size, "P.value" = ME2gores$result$p_value)


ggplot(data = dataME2GO, aes(x = "ME2", y = name,
                        color = P.value, size = ratio)) + 
  geom_point() +
  scale_color_gradient(low = "red", high = "blue") +
  theme_bw() + 
  ylab("") + 
  xlab("") + 
  ggtitle("GO enrichment analysis")

barplot(ME2gores, showCategory=20) 
ratio <- ME2gores$result$term_size
barplot(ratio ~ name, data=dataME2GO, horiz=TRUE)+
  scale_fill_gradient(low = "red", high = "blue") +
  ggtitle("GO enrichment analysis")

dataME2GO$name <- gsub("_", " ", dataME2GO$name)
ggplot(dataME2GO, aes(x = reorder(name,ratio), y = ratio, fill = P.value)) +
  geom_bar(stat = "identity", position = "dodge", ) +
  scale_fill_gradient(low = "red", high = "blue") +
  coord_flip() + 
  xlab("Gene ontology term") + 
  ylab("Ratio of GO terms present") +
  guides(fill=guide_legend(title="P value")) +
  theme(
    legend.position = c(.975, .05),
    legend.justification = c("right", "bottom"),
    legend.box.just = "right",
    legend.margin = margin(6, 6, 6, 6),
    plot.margin = unit(c(0,7,0,0),"cm")
    )
  

```

```{r}

make_module_heatmap <- function(module_name,
                                expression_mat = norm_counts,
                                metadata_df = coldata,
                                gene_module_key_df = gene_module_key,
                                module_eigengenes_df = module_eigengenes) {
  # Create a summary heatmap of a given module.
  #
  # Args:
  # module_name: a character indicating what module should be plotted, e.g. "ME19"
  # expression_mat: The full gene expression matrix. Default is `normalized_counts`.
  # metadata_df: a data frame with refinebio_accession_code and time_point
  #              as columns. Default is `metadata`.
  # gene_module_key: a data.frame indicating what genes are a part of what modules. Default is `gene_module_key`.
  # module_eigengenes: a sample x eigengene data.frame with samples as row names. Default is `module_eigengenes`.
  #
  # Returns:
  # A heatmap of expression matrix for a module's genes, with a barplot of the
  # eigengene expression for that module.

  # Set up the module eigengene with its refinebio_accession_code
  module_eigengene <- module_eigengenes_df %>%
    dplyr::select(all_of(module_name)) %>%
    tibble::rownames_to_column("refinebio_accession_code")

  # Set up column annotation from metadata
  col_annot_df <- metadata_df %>%
    # Only select the treatment and sample ID columns
    dplyr::select(refinebio_accession_code, Time, Treatment) %>%
    # Add on the eigengene expression by joining with sample IDs
    dplyr::inner_join(module_eigengene, by = "refinebio_accession_code") %>%
    # Arrange by patient and time point
    dplyr::arrange(Time, Treatment) %>%
    # Store sample
    tibble::column_to_rownames("refinebio_accession_code")

  # Create the ComplexHeatmap column annotation object
  col_annot <- ComplexHeatmap::HeatmapAnnotation(
    # Supply treatment labels
    Time = col_annot_df$Time,
    # Add annotation barplot
    module_eigengene = ComplexHeatmap::anno_barplot(dplyr::select(col_annot_df, module_name)),
    # Pick colors for each experimental group in time_point
    col = list(Treatment = colorlist)
  )

  # Get a vector of the Ensembl gene IDs that correspond to this module
  module_genes <- gene_module_key_df %>%
    dplyr::filter(module == module_name) %>%
    dplyr::pull(gene)

  # Set up the gene expression data frame
  mod_mat <- expression_mat %>%
    t() %>%
    as.data.frame() %>%
    # Only keep genes from this module
    dplyr::filter(rownames(.) %in% module_genes) %>%
    # Order the samples to match col_annot_df
    dplyr::select(rownames(col_annot_df)) %>%
    # Data needs to be a matrix
    as.matrix()

  # Normalize the gene expression values
  mod_mat <- mod_mat %>%
    # Scale can work on matrices, but it does it by column so we will need to
    # transpose first
    t() %>%
    scale() %>%
    # And now we need to transpose back
    t()

  # Create a color function based on standardized scale
  color_func <- circlize::colorRamp2(
    c(-2, 0, 2),
    c("#67a9cf", "#f7f7f7", "#ef8a62")
  )

  # Plot on a heatmap
  heatmap <- ComplexHeatmap::Heatmap(mod_mat,
    name = module_name,
    # Supply color function
    col = color_func,
    # Supply column annotation
    bottom_annotation = col_annot,
    # We don't want to cluster samples
    cluster_columns = FALSE,
    # We don't need to show sample or gene labels
    show_row_names = FALSE,
    show_column_names = FALSE
  )

  # Return heatmap
  return(heatmap)
}

mod_19_heatmap <- make_module_heatmap(module_name = "ME1")

```



```{r}



# extracting out genes from each module based on hypotheses previously formed and the module expression shape










options(stringsAsFactors = FALSE)

# Hypothesis #1: Peak symptom observation genes
#GHu WT 	7dpi – MEblue, MEpurple, MEbrown, MEyellow, MEblack
net$
blue_module <- tibble::enframe(MEs0, name = "gene", value = "module") %>%
  dplyr::mutate(module = paste0("ME",module))




probes = names(datExpr0)
names(datExpr0)[moduleColors=="blue"]


# Hypothesis #2: Recovery of GHu WT genes
# 12dpi – MElightcyan, MElightgreen


# F13 Mutant does not revert back the phenotype but created modules of interest with rather unique expression profiles at all three time points
#F13 Mutant 	4dpi – MEwhite, MEgreenyellow 

#7dpi – MEdarkturquoise, MEskyblue 

#12dpi – MEorange, MEdarkorange

# Hypothesis #3: A single amino acid disrupts plant-pathogen interactions to the extent of complete abolishment of symptomology.
#GHu Mutant	4dpi - MEcyan
#7dpi – MEdarkgreen, MEdarkred 



```


```{r}
WGCNA_input_gene <- data.frame(t(datExpr0))

WGCNA_input_gene <- data.frame(WGCNA_input_gene, module = moduleColors)                          
WGCNA_input_gene_filtered <- filter(WGCNA_input_gene, !module == "grey")
WGCNA_input_gene_filtered #25310 genes remained



cols <- c("grey"="grey","turquoise"="turquoise","blue"="blue","brown"="brown","yellow"="yellow","green"="green","red"="red","black"="black","pink"="pink","magenta"="magenta","purple"="purple","greenyellow"="greenyellow","tan"="tan","salmon"="salmon","cyan"="cyan","midnightblue"="midnightblue","lightcyan"="lightcyan","grey60"="grey60","lightgreen"="lightgreen","lightyellow"="lightyellow","royalblue"="royalblue","red"="red","green"="green","turquoise"="turquoise","grey"="grey") 
tiff('Module_trait relationships.tiff', units="in", width=4, height=6, res=1000, compression = 'lzw')

```
